packages/astro/src/runtime/client/dev-overlay/plugins/audit.ts

```diff
@@ -0,0 +1,119 @@
+
+interface AuditRule {
+	title: string;
+	message: string;
+}
+
+const selectorBasedRules: (AuditRule & { selector: string })[] = [
+	{
+		title: 'Missing `alt` tag',
+		message: 'The alt attribute is important for accessibility.',
+		selector: 'img:not([alt])',
+	},
+];
+
+export default {
+	id: 'astro:audit',
+	name: 'Audit',
+	icon: icon,
+	init(canvas, eventTarget) {
+		let audits: { highlightElement: DevOverlayHighlight; auditedElement: HTMLElement }[] = [];
+
+		selectorBasedRules.forEach((rule) => {
+			document.querySelectorAll(rule.selector).forEach((el) => {
+				createAuditProblem(rule, el);
+			});
+		});
+
+		if (audits.length > 0) {
+			eventTarget.dispatchEvent(
+				new CustomEvent('plugin-notification', {
+					detail: {
+						state: true,
+					},
+				})
+			);
+		}
+
+		function createAuditProblem(rule: AuditRule, originalElement: Element) {
+			const computedStyle = window.getComputedStyle(originalElement);
+			const targetedElement = (originalElement.children[0] as HTMLElement) || originalElement;
+
+			// If the element is hidden, don't do anything
+			if (targetedElement.offsetParent === null || computedStyle.display === 'none') {
+				return;
+			}
+
+			const highlight = document.createElement('astro-overlay-highlight') as DevOverlayHighlight;
```

Comment:
Same problem here as the islands overlay — no way to tab to and interact with the keyboard.